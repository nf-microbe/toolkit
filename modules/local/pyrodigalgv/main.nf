process PYRODIGALGV {
    tag "${meta.id}"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/pyrodigal-gv:0.3.2--pyh7e72e81_0':
        'biocontainers/pyrodigal-gv:0.3.2--pyh7e72e81_0' }"

    input:
    tuple val(meta), path(fasta)

    output:
    tuple val(meta), path("${prefix}.pyrodigalgv.gff.gz")   , emit: gff
    tuple val(meta), path("${prefix}.pyrodigalgv.faa.gz")   , emit: faa
    tuple val(meta), path("${prefix}.pyrodigalgv.fna.gz")   , emit: fna
    path "versions.yml"                                     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    pyrodigal-gv \\
        -i ${fasta} \\
        -a ${prefix}.pyrodigalgv.faa \\
        -d ${prefix}.pyrodigalgv.fna \\
        -f gff \\
        -o ${prefix}.pyrodigalgv.gff \\
        --jobs ${task.cpus} \\
        ${args}

    gzip ${prefix}.pyrodigalgv.faa ${prefix}.pyrodigalgv.fna ${prefix}.pyrodigalgv.gff

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        pyrodigalgv: \$(pyrodigal-gv --version | sed 's/pyrodigal-gv v//')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    echo "" | gzip > ${prefix}.pyrodigalgv.faa.gz
    echo "" | gzip > ${prefix}.pyrodigalgv.fna.gz
    echo "" | gzip > ${prefix}.pyrodigalgv.gff.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        pyrodigalgv: \$(pyrodigal-gv --version | sed 's/pyrodigal-gv v//')
    END_VERSIONS
    """
}
