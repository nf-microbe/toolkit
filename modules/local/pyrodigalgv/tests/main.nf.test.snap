{
    "sarscov2 - fasta.gz - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.pyrodigalgv.gff.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.pyrodigalgv.faa.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.pyrodigalgv.fna.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "3": [
                    "versions.yml:md5,82e5548ec8bb7c496c955aeaef6c13b2"
                ],
                "faa": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.pyrodigalgv.faa.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "fna": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.pyrodigalgv.fna.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "gff": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.pyrodigalgv.gff.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,82e5548ec8bb7c496c955aeaef6c13b2"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-11T09:27:23.741379651"
    },
    "sarscov2 - fasta.gz": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.pyrodigalgv.gff.gz:md5,664a230efa8f6adeebd10e2da8c63821"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.pyrodigalgv.faa.gz:md5,dc074252303061fc3533edb44b069338"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.pyrodigalgv.fna.gz:md5,b12dccd1f126ca157fbf18dca5994dd5"
                    ]
                ],
                "3": [
                    "versions.yml:md5,82e5548ec8bb7c496c955aeaef6c13b2"
                ],
                "faa": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.pyrodigalgv.faa.gz:md5,dc074252303061fc3533edb44b069338"
                    ]
                ],
                "fna": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.pyrodigalgv.fna.gz:md5,b12dccd1f126ca157fbf18dca5994dd5"
                    ]
                ],
                "gff": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.pyrodigalgv.gff.gz:md5,664a230efa8f6adeebd10e2da8c63821"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,82e5548ec8bb7c496c955aeaef6c13b2"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-11T09:27:00.988647594"
    }
}